File: Pythonleetcode/Leetcode119.py
class Solution(object):
    def getRow(self, rowIndex):
        def jiecheng(n):
            if n==0:
                return 1
            k=1
            for i in range(1,n+1):
                k=i*k
            return k


        ans=[1]*(rowIndex+1)

        for i in range(0, rowIndex):

            ans[i]=int(jiecheng(rowIndex)/(jiecheng(rowIndex-i)*jiecheng(i)))
        return ans

print(Solution().getRow(5))

File: CppPractice/CppLeetcode/Leetcode119.cpp
class Solution {
public:
    vector<int> getRow(int rowIndex) {
        if(rowIndex==0){return {1};}
        if(rowIndex==1){return {1,1};}
        vector<int> temp={1,1};
        for(int i=2;i<rowIndex+1;i++)
        {
            vector<int> newTemp = temp;
            newTemp.push_back(1);
            for(int j=1;j<newTemp.size()-1;j++){newTemp[j]=temp[j]+temp[j-1];}
            temp= newTemp;

        }
        return temp;

        
    }
};